### PR TITLE
Add support for get-root-token command

### PR DESCRIPTION
**Pull request by kamolhasan**
Signed-off-by: kamolhasan <kamol@appscode.com>

### CODE REVIEW

File: pkg/get-root-token/aws-kms-ssm/aws_kms_ssm.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func (ti *TokenInfo) TokenName() string {
+	sts, err := ti.kubeClient.AppsV1().StatefulSets(ti.vs.Namespace).Get(context.TODO(), ti.vs.Name, metav1.GetOptions{})
+	if err != nil {
+		return ""
+	}
+
+	var keyPrefix string
+	unsealerContainer := fmt.Sprintf("vault-%s", vaultapi.VaultUnsealerContainerName)
```
Comment (kamolhasan) [inline pkg/get-root-token/aws-kms-ssm/aws_kms_ssm.go]: Change:  vaultapi.VaultUnsealerContainerName = "vault-unsealer"

File: pkg/get-root-token/aws-kms-ssm/aws_kms_ssm.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+	secret, err := kubeClient.CoreV1().Secrets(vs.Namespace).Get(context.TODO(), vs.Spec.Unsealer.Mode.AwsKmsSsm.CredentialSecret, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if _, ok := secret.Data["access_key"]; !ok {
+		return nil, errors.Errorf("%s not found in secret", AccessKey)
+	}
+	if _, ok := secret.Data["secret_key"]; !ok {
```
Comment (kamolhasan) [inline pkg/get-root-token/aws-kms-ssm/aws_kms_ssm.go]: if ok {
if err = os.Setenv(SecretKey, value); err != nil {
		return nil, err
	}
}

File: pkg/get-root-token/azure-key-vault/azure_key_vault.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	if _, ok := secret.Data["client-id"]; !ok {
+		return nil, errors.Errorf("%s not found in secret", ClientID)
+	}
+	if _, ok := secret.Data["client-secret"]; !ok {
+		return nil, errors.Errorf("%s not found in secret", ClientSecret)
+	}
+
+	if err = os.Setenv(ClientID, string(secret.Data["client-id"])); err != nil {
```
Comment (kamolhasan) [inline pkg/get-root-token/azure-key-vault/azure_key_vault.go]: Do similar changes

File: pkg/get-root-token/azure-key-vault/azure_key_vault.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+func (ti *TokenInfo) TokenName() string {
+	sts, err := ti.kubeClient.AppsV1().StatefulSets(ti.vs.Namespace).Get(context.TODO(), ti.vs.Name, metav1.GetOptions{})
+	if err != nil {
+		return ""
+	}
+
+	var keyPrefix string
+	unsealerContainer := fmt.Sprintf("vault-%s", vaultapi.VaultUnsealerContainerName)
```
Comment (kamolhasan) [inline pkg/get-root-token/azure-key-vault/azure_key_vault.go]: Do similar change

File: pkg/get-root-token/google-kms-gcs/google_kms_gcs.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+	secret, err := kubeClient.CoreV1().Secrets(vs.Namespace).Get(context.TODO(), vs.Spec.Unsealer.Mode.GoogleKmsGcs.CredentialSecret, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if _, ok := secret.Data[ServiceAccountJSON]; !ok {
+		return nil, errors.Errorf("%s not found in secret", ServiceAccountJSON)
+	}
+
+	saFile := fmt.Sprintf("/tmp/%s", ServiceAccountJSON)
```
Comment (kamolhasan) [inline pkg/get-root-token/google-kms-gcs/google_kms_gcs.go]: Use random generated path: `/tmp/<google-sa-cred-6 char random>/sa.json`

File: pkg/get-root-token/google-kms-gcs/google_kms_gcs.go [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+	if err = os.Setenv(GoogleApplicationCred, saFile); err != nil {
+		return nil, err
+	}
+
+	client, err := storage.NewClient(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+
+	return &TokenInfo{
```
Comment (kamolhasan) [inline pkg/get-root-token/google-kms-gcs/google_kms_gcs.go:97]: Store the path info here

File: pkg/get-root-token/google-kms-gcs/google_kms_gcs.go [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+		storageClient: client,
+		kubeClient:    kubeClient,
+		vs:            vs,
+	}, nil
+}
+
+func (ti *TokenInfo) Token() (string, error) {
+	token := ti.TokenName()
```
Comment (kamolhasan) [inline pkg/get-root-token/google-kms-gcs/google_kms_gcs.go:110]: Cleanup the path using `defer`

File: pkg/get-root-token/google-kms-gcs/google_kms_gcs.go [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+func (ti *TokenInfo) TokenName() string {
+	sts, err := ti.kubeClient.AppsV1().StatefulSets(ti.vs.Namespace).Get(context.TODO(), ti.vs.Name, metav1.GetOptions{})
+	if err != nil {
+		return ""
+	}
+
+	var keyPrefix string
+	unsealerContainer := fmt.Sprintf("vault-%s", vaultapi.VaultUnsealerContainerName)
```
Comment (kamolhasan) [inline pkg/get-root-token/google-kms-gcs/google_kms_gcs.go]: Do the similar changes

File: pkg/get-root-token/kubernetes-secret/kubernetes_secret.go [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+func (ti *TokenInfo) Token() (string, error) {
+	secretName := ti.vs.Spec.Unsealer.Mode.KubernetesSecret.SecretName
+	secretNamespace := ti.vs.Namespace
+	secret, err := ti.kubeClient.CoreV1().Secrets(secretNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	token := ti.TokenName()
```
Comment (kamolhasan) [inline pkg/get-root-token/kubernetes-secret/kubernetes_secret.go:66]: Do similar changes

File: pkg/get-root-token/kubernetes-secret/kubernetes_secret.go [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+func (ti *TokenInfo) TokenName() string {
+	sts, err := ti.kubeClient.AppsV1().StatefulSets(ti.vs.Namespace).Get(context.TODO(), ti.vs.Name, metav1.GetOptions{})
+	if err != nil {
+		return ""
+	}
+
+	var keyPrefix string
+	unsealerContainer := fmt.Sprintf("vault-%s", vaultapi.VaultUnsealerContainerName)
```
Comment (kamolhasan) [inline pkg/get-root-token/kubernetes-secret/kubernetes_secret.go]: Do similar changes

File: pkg/cmds/get_root_token.go [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+	"k8s.io/client-go/kubernetes"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+func NewCmdGetRootToken(clientGetter genericclioptions.RESTClientGetter) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "get-root-token",
+		Short:             "Get root token for vault server",
+		DisableAutoGenTag: true,
```
Comment (kamolhasan) [inline pkg/cmds/get_root_token.go:59]: Add long description

**Review by sakiib — COMMENTED** [review]
LGTM!

**Review by sakiib — COMMENTED** [review]
LGTM!

**Review by kamolhasan — COMMENTED** [review]
LGTM!

**Review by sakiib — APPROVED** [review]

**Pull request merged**